api/gateway/validation.go

```diff
@@ -23,14 +23,30 @@ func (m *Status) Validate() error {
 	return nil
 }
 
+// Location metadata errors
+var (
+	ErrInvalidLatitude  = errors.NewErrInvalidArgument("LocationMetadata", "invalid latitude")
+	ErrInvalidLongitude = errors.NewErrInvalidArgument("LocationMetadata", "invalid longitude")
+	ErrLocationZero     = errors.NewErrInvalidArgument("LocationMetadata", "is zero, so should be nil")
+)
+
 // Validate implements the api.Validator interface
-func (m *GPSMetadata) Validate() error {
-	if m == nil || m.IsZero() {
-		return errors.NewErrInvalidArgument("GPSMetadata", "can not be empty")
+func (m LocationMetadata) Validate() error {
+	if m.IsZero() {
+		return ErrLocationZero
+	}
+	if m.Latitude >= 90-delta || m.Latitude <= -90+delta {
+		return ErrInvalidLatitude
+	}
+	if m.Longitude >= 180-delta || m.Longitude <= -180+delta {
```

Comment:
I don't think we should add the delta for the longitude. The chance is much higher for a gateway standing on or close to the -180/180 (international date) line. For -90/+90 it is a point (the poles), for which the probability is extremely low.